modules/nw-sriov-example-dpdk-line-rate.adoc

```diff
@@ -0,0 +1,278 @@
+     nodeSelector:
+     node-role.kubernetes.io/worker-cnf: ""
+     numVfs: 10
+     priority: 99
+     resourceName: dpdk_nic_2
+
+----
+<1> For Intel NICs, `deviceType` must be `vfio-pci`.
+<2> If kernel communication with DPDK workloads is required, add `needVhostNet: true`. This will mount the `/dev/net/tun` and `/dev/vhost-net` devices into the container so the application can create a tap device and connect the tap device to the DPDK workload.
+
+The following is an example of `sriovNetworkNodePolicy` for Mellanox NICs:
+[source,yaml]
+----
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetworkNodePolicy
+    metadata:
+     name: dpdk-nic-1
+     namespace: openshift-sriov-network-operator
+    spec:
+     deviceType: netdevice <1>
+     isRdma: true <2>
+     nicSelector:
+       rootDevices:
+         - "0000:5e:00.1"
+     nodeSelector:
+       node-role.kubernetes.io/worker-cnf: ""
+     numVfs: 5
+     priority: 99
+     resourceName: dpdk_nic_1
+    ---
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetworkNodePolicy
+    metadata:
+     name: dpdk-nic-2
+     namespace: openshift-sriov-network-operator
+    spec:
+     deviceType: netdevice
+     isRdma: true
+     nicSelector:
+       rootDevices:
+         - "0000:5e:00.0"
+     nodeSelector:
+       node-role.kubernetes.io/worker-cnf: ""
+     numVfs: 5
+     priority: 99
+     resourceName: dpdk_nic_2
+----
+<1> For Mellanox devices the `deviceType` must be `netdevice`.
+<2> For Mellanox devices `isRDMA` must be `true`.
+Mellanox cards are connected to DPDK applications using Flow Bifurcation. This mechanism splits traffic between Linux user space and kernel space, and can enhance line rate processing capability.
+
+== Using the SR-IOV network operator to create the `sriovNetwork` object
+The following is an example definition of an `sriovNetwork` object. In this case Intel and Mellanox configurations are identical.
+[source,yaml]
+----
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetwork
+    metadata:
+     name: dpdk-network-1
+     namespace: openshift-sriov-network-operator
+    spec:
+     ipam: '{"type": "host-local","ranges": [[{"subnet": "10.0.1.0/24"}]],"dataDir":
+       "/run/my-orchestrator/container-ipam-state-1"}' <1>
+     networkNamespace: dpdk-test <2>
+     spoofChk: "off"
+     trust: "on"
+     resourceName: dpdk_nic_1 <3>
+    ---
+    apiVersion: sriovnetwork.openshift.io/v1
+    kind: SriovNetwork
+    metadata:
+     name: dpdk-network-2
+     namespace: openshift-sriov-network-operator
+    spec:
+     ipam: '{"type": "host-local","ranges": [[{"subnet": "10.0.2.0/24"}]],"dataDir":
+       "/run/my-orchestrator/container-ipam-state-1"}'
+     networkNamespace: dpdk-test
+     spoofChk: "off"
+     trust: "on"
+     resourceName: dpdk_nic_2
+----
+<1> You can use a different IPAM, such as Whereabouts. See "Dynamic IP address assignment configuration with Whereabouts" for more information.
```

Comment:
For consistency with other references in the library, you could update the second sentence as follows:

> For more information, see _Dynamic IP address assignment configuration with Whereabouts_.

Reply:
Done.